src/middlewared/middlewared/plugins/disk_/wipe.py

```diff
@@ -0,0 +1,100 @@
+import asyncio
+import os
+import platform
+import re
+import signal
+import subprocess
+
+from middlewared.schema import accepts, Bool, Str
+from middlewared.service import job, private, Service
+from middlewared.utils import Popen, run
+
+
+RE_DD = re.compile(r'^(\d+) bytes transferred .*\((\d+) bytes')
+IS_LINUX = platform.system().lower() == 'linux'
+
+
+class DiskService(Service):
+
+    @private
+    async def destroy_partitions(self, disk):
+        if IS_LINUX:
+            await run(['sgdisk', '-Z', os.path.join('/dev', disk)])
+        else:
+            await run('gpart', 'destroy', '-F', f'/dev/{disk}', check=False)
+            # Wipe out the partition table by doing an additional iterate of create/destroy
+            await run('gpart', 'create', '-s', 'gpt', f'/dev/{disk}')
+            await run('gpart', 'destroy', '-F', f'/dev/{disk}')
+
+    @private
+    async def wipe_quick(self, dev, size=None):
+        # If the size is too small, lets just skip it for now.
+        # In the future we can adjust dd size
+        if size and size < 33554432:
+            return
+        await run('dd', 'if=/dev/zero', f'of=/dev/{dev}', 'bs=1M', 'count=32')
+        size = await self.middleware.call('device.get_dev_size', dev)
+        if not size:
+            self.logger.error(f'Unable to determine size of {dev}')
+        else:
+            # This will fail when EOL is reached
+            await run('dd', 'if=/dev/zero', f'of=/dev/{dev}', 'bs=1m', f'oseek={int(size / 1024) - 32}', check=False)
```

Comment:
I dont think 1m lowercase works in linux?

Reply:
It doesn't, i must have lost my changes during the rebase. Going to QA it again after addressing reviews :)